rpcserver.go

```diff
@@ -4347,10 +4347,14 @@ func (r *rpcServer) extractPaymentIntent(rpcPayReq *rpcPaymentRequest) (rpcPayme
 		return payIntent, errors.New("invalid payment address length")
 	}
 
-	if payIntent.paymentAddr == nil {
+	// Set the payment address if it was explicitly defined with the
+	// rpcPaymentRequest.
+	// Note that the payment address for the payIntent should be nil if none
+	// was provided with the rpcPaymentRequest.
+	if len(rpcPayReq.PaymentAddr) != 0 {
 		payIntent.paymentAddr = &[32]byte{}
+		copy(payIntent.paymentAddr[:], rpcPayReq.PaymentAddr)
 	}
-	copy(payIntent.paymentAddr[:], rpcPayReq.PaymentAddr)
```

Comment:
Ahh, so this ended up copying a _zero valued_ payment address, which was rejected as it's used as a shibboleth elsewhere in the codebase.